packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -441,6 +489,32 @@ export class FirestoreRepository implements Repository {
     return customElementsResults.docs.map((d) => d.data());
   }
 
+  async queryElements({
+    query,
+    limit,
+  }: {
+    query?: string;
+    limit?: number;
+  }): Promise<Array<CustomElement>> {
+    let dbQuery: Query<CustomElement> | CollectionGroup<CustomElement> = db
+      .collectionGroup('customElements')
+      .withConverter(customElementConverter)
+      .where('isLatest', '==', true)
+      .limit(limit ?? 25);
+
+    if (query !== undefined) {
+      // Split query
+      const queryTerms = natural.PorterStemmer.tokenizeAndStem(query);
+      if (queryTerms.length > 10) {
+        queryTerms.length = 10;
+      }
+      dbQuery = dbQuery.where('searchTerms', 'array-contains-any', queryTerms);
+    }
+
```

Comment:
File an issue for ranking and add TODO? As discussed offline, I think a word frequency table (computed from our own data) with a https://en.wikipedia.org/wiki/Tf%E2%80%93idf score would be a really simple and good start.

Reply:
https://github.com/webcomponents/webcomponents.org/issues/1361